sabnzbd/downloader.py

```diff
@@ -387,10 +393,7 @@ def highest_server(self, me: Server):
         """Return True when this server has the highest priority of the active ones
         0 is the highest priority
         """
-        for server in self.servers:
-            if server is not me and server.active and server.priority < me.priority:
-                return False
-        return True
+        return me.priority == self.servers[0].priority
```

Comment:
I don't fully get it, why just comparing to the first one is enough?
Does that work when there's 3 levels of priority?

Reply:
The way I understand this function, it should return false if there is a higher level server. It doesn't matter how much higher. It is used for stopping lower level servers from running ahead of the top level servers and loading articles from disk that they won't be allowed to try to download.

If it is a top level server and the list is unsorted it has to go through the whole list to know that there are no higher servers. When it's sorted, a top level server will be placed in index 0, so it must have the same priority as that one. If the priority number is different, it can't be a top level server. It doesn't matter how many levels there are because all the servers that aren't top level should return false.

Reply:
You are right, I forgot what this function does. Just postpone the loading of articles to at least wait when we tried the first level servers.